### ISSUE TITLE
Control date range of data pulls for applicable streams

### ISSUE THREAD

**Issue by jlloyd-widen** [AutoIDM/tap-googleads#29]
This is a new feature request. It would be good to be able to control the number of days pulled from Google Ads in certain taps. Most of time this is controlled in the Google Ads query with something like `WHERE segments.date DURING LAST_7_DAYS`. Changing to a `segments.date BETWEEN '2019-01-01' AND '2019-01-31'` strategy controlled by the configuration parameters `start_date` and `end_date` would be ideal.

**Comment by visch** [comment]
Yes that would be much better, even better if we had incremental replication setup.  Hard coding the last 7 days isn't the best  :/ 

Not sure what the limits are to history and how that works with Google Ads specifically.

**Comment by visch** [comment]
We don't have State / Incremental replication but we do have History available now which solves the core issue! I'll leave this open but it's now just for implementing the State functionality of Singer
